mock-kubectl

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+[[ -n $DEBUG ]] && set -x
+
+set -eou pipefail
+
+if hash kubectl 2>/dev/null; then
```

Comment:
I don't understand why this code block is here. 

I think we should override KUBECTL only for calls that fail without a cluster (like `kubectl get namespace`); not the entire test suite.

that would simplify both the mock-kubectl script, and makes it easier to reason about the tests.

Reply:
removed all this junk. it was unnecessary after all.